infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/ClusterDescriptor.java

```diff
@@ -13,70 +11,20 @@
  */
 public class ClusterDescriptor {
 
-    private static int CORFU_PORT = 9000;
-
-    @Getter
-    String clusterId;
-
     @Getter
-    ClusterRole role;
+    final String clusterId;
 
     @Getter
-    List<NodeDescriptor> nodesDescriptors;
+    final List<NodeDescriptor> nodeDescriptors;
 
     @Getter
-    private int corfuPort;    // Port on which Corfu DB runs on this cluster
-
-    public ClusterDescriptor(String clusterId) {
-        this.clusterId = clusterId;
-    }
-
-    public ClusterDescriptor(ClusterConfigurationMsg clusterConfig) {
-        this.clusterId = clusterConfig.getId();
-        this.role = clusterConfig.getRole();
-        this.corfuPort = clusterConfig.getCorfuPort() != 0 ? clusterConfig.getCorfuPort() : CORFU_PORT;
-        this.nodesDescriptors = new ArrayList<>();
-        for (NodeConfigurationMsg nodeConfig : clusterConfig.getNodeInfoList()) {
-            NodeDescriptor newNode = new NodeDescriptor(nodeConfig.getAddress(),
-                    Integer.toString(nodeConfig.getPort()), clusterId,
-                    nodeConfig.getConnectionId(),
-                    nodeConfig.getNodeId());
-            this.nodesDescriptors.add(newNode);
-        }
-    }
-
-    public ClusterDescriptor(ClusterDescriptor info, ClusterRole roleType) {
-        this.clusterId = info.clusterId;
-        this.role = roleType;
-        this.nodesDescriptors = new ArrayList<>();
-        this.corfuPort = info.getCorfuPort();
-        for (NodeDescriptor nodeInfo : info.nodesDescriptors) {
-            NodeDescriptor newNode = new NodeDescriptor(nodeInfo.getHost(), nodeInfo.getPort(),
-                    info.clusterId, nodeInfo.getConnectionId(), nodeInfo.getNodeId());
-            this.nodesDescriptors.add(newNode);
-        }
-    }
+    private final int corfuPort;    // Port on which Corfu DB runs on this cluster
 
-    public ClusterDescriptor(String clusterId, ClusterRole roleType, int corfuPort) {
+    @VisibleForTesting
+    public ClusterDescriptor(String clusterId, int corfuPort, List<NodeDescriptor> nodes) {
```

Comment:
if this is anyway the exposed constructor, do we still need to annotate it with VisibleForTesting?

Reply:
yeah, it serves as a marker for us as well that it is used for testing. Makes the code easy to read